hm-module/portal.nix

```diff
@@ -0,0 +1,118 @@
+{ self, lib, ... }:
+{ config, pkgs, ... }:
+let
+  inherit (lib) types;
+
+  cfg = config.wayland.windowManager.hyprland.portal;
+
+  # If the overlay is applied to Nixpkgs, `xdg-desktop-portal-hyprland` in
+  # `pkgs` is probably newer. Otherwise, the package provided by the flake is
+  # guaranteed to be newer.
+  defaultPackage = let
+    inNixpkgs = pkgs.xdg-desktop-portal-hyprland.version;
+    inFlake = self.packages.${pkgs.system}.xdg-desktop-portal-hyprland.version;
+  in if lib.versionOlder inNixpkgs inFlake then
+    pkgs.xdg-desktop-portal-hyprland
+  else
+    self.packages.${pkgs.system}.xdg-desktop-portal-hyprland;
+in {
+  options = {
+    wayland.windowManager.hyprland.portal = {
+      enable = lib.mkOption {
+        type = types.bool;
+        example = true;
+        description = ''
+          Whether to enable {option}`xdg.portal.enable` and configure
+          to use {package}`xdg-desktop-portal-hyprland`.
```

Comment:
Almost missed this, `package` is not a valid role so something like nixos-render-docs would error our while ndg converts this verbatim.